File: src/main/java/de/whitefrog/neobase/model/annotation/RelationshipCount.java
package de.whitefrog.neobase.model.annotation;

import de.whitefrog.neobase.model.Model;
import org.neo4j.graphdb.Direction;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface RelationshipCount {
  String type() default "None";
  Direction direction() default Direction.OUTGOING;
  Class<? extends Model> otherModel() default Model.class;
}

File: src/main/kotlin/de/whitefrog/neobase/model/annotation/RelationshipCount.kt
package de.whitefrog.neobase.model.annotation

import de.whitefrog.neobase.model.Model
import org.neo4j.graphdb.Direction
import kotlin.reflect.KClass

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class RelationshipCount(val type: String = "None", val direction: Direction = Direction.OUTGOING, val otherModel: KClass<out Model> = Model::class)
